### ISSUE TITLE
Implement ability to import * from a package

### ISSUE THREAD

**Issue by nathanielc** [influxdata/flux#2741]
Our general plan for introducing new Flux is to move the packages around and let the user decide how they want to interact with Flux.  See #2740 for moving the existing packages.

Once we have two packages the user should be able to `import *` from which ever package they want and it will define how they want to use Flux.

For example a user who wants to continue to use Flux as it stands today could do something like this:

```
import * from "schemaless"

from(bucket:"foo")
    |> range()
    |> filter()
```

A user wanting to use the new Flux could do this:

```
import * from "table"

@foo.temperatures{state,location}
    |> aggregate({min(bottom_degrees)}, by:["location"])
```

After #2740 is complete the prelude will have very little content so its likely users will need to specify which one they prefer. Additionally we can change the prelude between Flux versions to change the effective default.

Other considerations...

What should the syntax be?

The above examples are similar to the python syntax, but our convention so far is to follow Go's import syntax which is `import . "table"`. What syntax do we like?

We need to specify the semantics. 

The basics are that importing * brings all identifies from the package into the current package scope. There are some edge cases we need to define: 
* How do we resolve conflicts with current scope?
* How do we resolve conflicts with other packages?

**Comment by nathanielc** [comment]
NOTES:

* The decision is that we use the `import . "foo"` syntax
* When importing symbols from a package the are added to the current package scope. As such any reassignment from other `import .` statements or local variable assignments will be seen as reassignments and produce an error.
* The question of visibility (i.e. public vs private) are not part of this issue. All symbols should be imported.


See SPEC for description of scopes https://github.com/influxdata/flux/blob/master/docs/SPEC.md#assignment-and-scope. Importing a package into local scope does not add a new scoping layer but instead adds those identifiers to the existing file block scope.

**Comment by jsternberg** [comment]
This issue may need to be split up. Each of the individual steps for this may not be too difficult, but in a conversation we identified a lot of moving parts that, together, each may end up taking a day or two by themselves.

1. Modify the parser to be able to recognize the new syntax and figure out how to annotate it. Do we set the name as `.` for the package or add a new field to the AST tree to signify the import?
2. Did the AST above change and we need to add that change to the Go AST and the Rust AST if it did. If it didn't, there might not be work here. The `import .` statement may also be a separate statement type because the current import doesn't add more than one variable to the scope but `import .` acts differently by introducing multiple variables to the scope. This effects whether a few of the below tasks even need to be done. If we set the name of the package as `.` it makes it easier to introduce this syntax, but it also potentially causes us to accidentally interpret the `.` as a real name if we forget to add a special check code in one place of the code base that uses the package name.
3. Same question for the semantic graph.
4. Type inference needs to be updated to recognize the new syntax and return errors when a conflict happens.
5. We might need to encode the change to the semantic graph to flatbuffers.
6. The Go code for imports need to be changed.

I guess the first question is about how `import .` should be represented in the AST because the estimate will be wildly different depending on the answer to that.
